### DISCUSSION TITLE
Change default value for "number of servers to query simultaneously"?

### DISCUSSION THREAD

**Discussion by The-Gig** [torhus/monster-browser#31] [Ideas]
Hi!
I've done a few tests with different values for "number of servers to query simultaneously" parameter.

I've done a "refresh all" for "quake iii (all servers)" (about 588 servers) with MB beta Oct 11 2022.
- set to 1: 1 min 43 sec
- set to 5: 22 sec
- set to 10 (current default): 11 sec
- set to 20, 50 or 99: 7-8 sec

Note: 99 is current max allowed by the GUI, as if you set an higher value it reverts to 10.

With it set to 10, it seems to take up about 500 kbit/s download bandwidth. With it set to 20, 50 or 99, it seems to take up about 
800 kbit/s.

So, maybe it may be worth it to change the default value from 10 to 20 or so?

Testing with "urban terror 4.3" (about 234 servers):
- set to 10: 5 sec
- set to 20, 50 or 99: 4 sec

**Comment by torhus** [comment]
It's set to 10 to reduce congestion. If you set it higher, more servers will time out, or get a much higher ping than they really should. I have a 100 Mb/s connection, which many people don't. And I still see it.

Ideally, you would ping the servers first, then query them for information. I think that's what The All-Seeing Eye does.

**Reply by The-Gig** [reply]
The All-Seeing Eye! Wow, how many years passed since I used it... Does it still work?

**Reply by torhus** [reply]
Doesn't look like it: https://en.wikipedia.org/wiki/The_All-Seeing_Eye

**Reply by The-Gig** [reply]
Exactly, I knew it was gone since a looong time. For this reason the fact you said "does" instead of "did" caught me off-guard... :)

**Reply by torhus** [reply]
Yeah, I wasn't sure if it still works :)

**Comment by The-Gig** [comment]
Anyway, what about adding in readme.txt something like "Number of server to query simultaneously is set to 10 by default. Lower values might slow down refresh a lot, while higher values might speed up refresh a little, but also could end up showing higher pings than reality and even a few timeouts."?

**Reply by torhus** [reply]
I probably should, now it's issue #36!
